lib/min_heap.rb

```diff
@@ -8,26 +8,37 @@ def initialize(key, value)
 end
 
 class MinHeap
-
   def initialize
     @store = []
   end
 
   # This method adds a HeapNode instance to the heap
-  # Time Complexity: ?
-  # Space Complexity: ?
+  # Time Complexity: O(logn) where n is the length of the @store
+  # Space Complexity: O(1)
   def add(key, value = key)
-    raise NotImplementedError, "Method not implemented yet..."
+    new_node = HeapNode.new(key, value)
+    @store.push(new_node)
+
+    if @store.length > 1
+      heap_up((@store.length - 1))
+    end
   end
 
   # This method removes and returns an element from the heap
   #   maintaining the heap structure
-  # Time Complexity: ?
-  # Space Complexity: ?
+  # Time Complexity: O(logn) where n is the length of the @store
+  # Space Complexity: O(1)
```

Comment:
Since your `heap_down` method is recursive you have O(log n) being saved on the call stack.